SimPEG/utils/model_utils.py

```diff
@@ -260,3 +260,77 @@ def ismember(a, b):
     actv[inds] = True
 
     return actv
+
+
+def depth_weighting(mesh, reference_locs, indActive=None, exponent=None, threshold=None):
+
+    """
+    Create depth weighting
+
+    Parameters
+    ----------
+    mesh : Mesh object
+        discretize model space.
+    reference_locs : receiver locations
+        float or np.ndarray with shape of (points, dim).
+    indActive : array of bool
+        index vector for the active cells on the mesh below the topography.
+    exponent : float
+        exponent parameter for depth weighting.
+    threshold : float
+        adjustable constant parameter.
+
+    Returns
+    -------
+    wz : numpy.ndarray
+        normazlied depth weighting captures decay of the potential field data.
+
+    Notes
+    -----
+
+    ..math::
+
+        w(z) = (delta_z + threshold) ** (-0.5 * exponent)
+
+        where delta_z is depth of model cells along the z direction from receiver locations;
+        exponent and threshold are two adjustable parameters.
+
+
+      """
+
+
+    # Default exponent value
+    if exponent is None:
+        exponent = 2.0
+
+    # Default threshold value
+    if threshold is None:
+        threshold = 0.5 * mesh.h_gridded.min()
+
+    assert isinstance(reference_locs, (float, np.ndarray))
```

Comment:
do not use asserts,
also this is way too limiting, best idea would probably be to just do:
`reference_locs = np.asarray(reference_locs)`

Then further down (line 314) just do
`if reference_locs.ndim < 2:`

Reply:
if '''reference_locs.ndim == 1''', '''reference_locs''' is a 1d array. 
But seems we cannot do '''mesh.cell_centers[:,-1] - reference_locs''' if they have different size.

So I just do:
'''
if reference_locs.ndim == 0:
'''

Reply:
That’s why I suggested doing `ndim < 2` also if you try it, there’s a few different cases where it would still work as you’ve written it, if it’s a 0d array, a 1d array with one element, or a 1d array of the same length as the number of cells.

Reply:
I see, thanks!